src/test/java/com/acrolinx/sidebar/utils/StartPageInstallerTest.java

```diff
@@ -91,47 +83,4 @@ public void prepareSidebarUrlWithStartPageIfServerUrlIsSet() throws Exception
 
         assertEquals(StartPageInstaller.prepareSidebarUrl(params), StartPageInstaller.getStartPageURL());
     }
-
```

Comment:
Why was this test removed?

Reply:
Because it was useless and misleading. The method name said it was testing "from different threads", but that is not what the code actually did. Three Runnable instances were created and then .run() was called on each of them. This does not launch any new thread, all three runnables are executed on the same thread which runs the unit test method. To start new threads Thread.start(...) has to be used. Looks like the author of this code did not understand how Threads in Java are working.

Starting multiple threads in a unit test is usually a code smell and something which should be avoided, since it often leads to race conditions and hard to maintain code (e.g. a failing Assert.assertSomething(...) on a different thread will not cause the unit test to fail since the thread which executes the unit test method is not informed that an assert has failed in a concurrent thread).

Therefore rather than trying to "fix" this test I decided to remove it completely.

Reply:
Great! 🤓